pkg/manifests/manifests.go

```diff
@@ -0,0 +1,193 @@
+
+	for _, manifestPath := range combinedManifestPaths {
+		manifestData, err := readManifest(manifestPath)
+		if err != nil {
+			return fmt.Errorf("error reading manifest %w", err)
+		}
+
+		foundImages, err := findImagesInManifest(manifestData)
+		if err != nil {
+			return fmt.Errorf("error finding images in manifest %w", err)
+		}
+		extractedImages = append(extractedImages, foundImages...)
+	}
+
+	extractedImages = removeDuplicateImages(extractedImages)
+	addImagesToRegistryDefinition(ctx, extractedImages)
+
+	return nil
+}
+
+func readManifest(manifestPath string) (interface{}, error) {
+	manifestData, err := os.ReadFile(manifestPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading manifest: %w", err)
+	}
+
+	if len(manifestData) == 0 {
+		return nil, fmt.Errorf("invalid manifest")
+	}
+
+	var manifest interface{}
+	err = yaml.Unmarshal(manifestData, &manifest)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling manifest YAML: %w", err)
+	}
+	return manifest, nil
+}
+
+func findImagesInManifest(data interface{}) ([]string, error) {
+	imageSet := make(map[string]struct{})
+
+	var findImages func(data interface{})
+	findImages = func(data interface{}) {
+		switch t := data.(type) {
+		case map[string]interface{}:
+			for k, v := range t {
+				if k == "image" {
+					if imageName, ok := v.(string); ok {
+						imageSet[imageName] = struct{}{}
+					}
+				}
+				findImages(v)
+			}
+		case []interface{}:
+			for _, v := range t {
+				findImages(v)
+			}
+		}
+	}
+
+	findImages(data)
+
+	images := make([]string, 0, len(imageSet))
+	for uniqueImage := range imageSet {
+		images = append(images, uniqueImage)
+	}
+
+	return images, nil
+}
+
+func copyManifests(src string, dest string) ([]string, error) {
+	if dest == "" {
+		return nil, fmt.Errorf("manifest destination directory not defined")
+	}
+
+	var manifestPaths []string
+
+	manifests, err := os.ReadDir(src)
+	if err != nil {
+		return nil, fmt.Errorf("reading manifest source dir: %w", err)
+	}
+
+	for _, manifest := range manifests {
+		manifestName := strings.ToLower(manifest.Name())
+		if filepath.Ext(manifestName) != ".yaml" && filepath.Ext(manifestName) != ".yml" {
+			zap.S().Warnf("Skipping %s as it is not a yaml file", manifest.Name())
+			continue
+		}
+
+		sourcePath := filepath.Join(src, manifest.Name())
+		destPath := filepath.Join(dest, manifest.Name())
+		err := fileio.CopyFile(sourcePath, destPath, fileio.NonExecutablePerms)
+		if err != nil {
+			return nil, fmt.Errorf("copying manifest file %s: %w", sourcePath, err)
+		}
+		manifestPaths = append(manifestPaths, destPath)
+
+	}
+
+	return manifestPaths, nil
+}
+
+func downloadManifests(ctx *image.Context, destPath string) ([]string, error) {
+	manifests := ctx.ImageDefinition.Kubernetes.Manifests.URLs
+	var manifestPaths []string
+
+	for index, manifestURL := range manifests {
+		filePath := filepath.Join(destPath, fmt.Sprintf("manifest-%d.yaml", index+1))
```

Comment:
Is this naming the final decision we set for?

Reply:
No, that's up for discussion, just putting that temporarily.